cfgmgr/schedulermgr.cpp

```diff
@@ -0,0 +1,189 @@
+                                                                                                         m_stateTimeRangeStatusTable(stateDb, STATE_TIME_RANGE_STATUS_TABLE_NAME)
+{
+}
+
+task_process_status SchedulerMgr::writeCrontabFile(const string &fileName, const string &schedule, const string &command, bool deleteSelfAfterCompletion)
+{
+    string cronFileName = CRON_FILES_PATH_PREFIX_STR + fileName;
+    ofstream crontabFile{cronFileName};
+
+    if (crontabFile.fail())
+    {
+        SWSS_LOG_ERROR("Failed to create crontab file for %s", fileName.c_str());
+        return task_process_status::task_need_retry;
+    }
+    crontabFile << schedule << " ";
+    crontabFile << CRON_USERNAME_STR << " ";
+    crontabFile << command;
+    if (deleteSelfAfterCompletion)
+    {
+        crontabFile << " && rm " << cronFileName;
+    }
+    crontabFile << endl;
+    crontabFile.close();
+
+    SWSS_LOG_DEBUG("Crontab file for %s has been created", fileName.c_str());
+    return task_process_status::task_success;
+}
+
+// TODO add rollback mechanism
+task_process_status SchedulerMgr::createCronjobs(const string &taskName, const string &startTime, const string &endTime, bool runOnce)
+{
+    string enableCrontabName = taskName + "-enable";
+    string disableCrontabName = taskName + "-disable";
+
+    // Create command for enabling the task
+    string command_enabled;
+    {
+        stringstream ss;
+        ss << "/usr/bin/redis-cli -n " << STATE_DB << " HSET '" << STATE_TIME_RANGE_STATUS_TABLE_NAME << "|" << taskName << "' '" << TIME_RANGE_STATUS_STR << "' '" << TIME_RANGE_ENABLED_STR << "'";
+        command_enabled = ss.str();
+    }
+
+    // Create command for disabling the task
+    string command_disabled;
+    {
+        stringstream ss;
+        ss << "/usr/bin/redis-cli -n " << STATE_DB << " HSET '" << STATE_TIME_RANGE_STATUS_TABLE_NAME << "|" << taskName << "' '" << TIME_RANGE_STATUS_STR << "' '" << TIME_RANGE_DISABLED_STR << "'";
+        if (runOnce){
+            // Delete the time range configuration entry after the task has been disabled
+            // writeCrontabFile() will delete the crontab file itself after the task has been executed
+            ss << " && /usr/bin/redis-cli -n " << CONFIG_DB << " del '" << CFG_TIME_RANGE_TABLE_NAME << "|" << taskName << "'";
+        }
+        command_disabled = ss.str();
+    }
+
+    // Service file for enabling the task
+    if (writeCrontabFile(enableCrontabName, startTime, command_enabled, runOnce) != task_process_status::task_success)
+    {
+        return task_process_status::task_need_retry;
+    }
+
+    // Service file for disabling the task
+    if (writeCrontabFile(disableCrontabName, endTime, command_disabled, runOnce) != task_process_status::task_success)
+    {
+        return task_process_status::task_need_retry;
+    }
+
+    SWSS_LOG_INFO("Succesfully created crontab files for %s", taskName.c_str());
```

Comment:
This log already exists inside  writeCrontabFile

Reply:
The log in writeCrontabFile is DEBUG/ERROR logs, this one is an INFO log. Also they are for a single file, as opposed to this log which specified both files have been successfully created.